sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/CosmosSourceTaskConfig.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CosmosSourceTaskConfig extends CosmosSourceConfig {
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+    private static final String SOURCE_TASK_CONFIG_PREFIX = "kafka.connect.cosmos.source.task.";
+
+    public static final String SOURCE_METADATA_TASK_UNIT = SOURCE_TASK_CONFIG_PREFIX + "metadataTaskUnit";
+    public static final String SOURCE_FEED_RANGE_TASK_UNITS = SOURCE_TASK_CONFIG_PREFIX + "feedRangeTaskUnits";
+
+    private final List<FeedRangeTaskUnit> feedRangeTaskUnits;
+    private MetadataTaskUnit metadataTaskUnit;
+
+    public CosmosSourceTaskConfig(Map<String, String> parsedConfigs) {
+        super(getConfigDef(), parsedConfigs);
+
+        this.feedRangeTaskUnits = this.parseFeedRangeTaskUnits();
+        this.metadataTaskUnit = this.parseMetadataTaskUnit();
+    }
+
+    public static ConfigDef getConfigDef() {
+        ConfigDef configDef = CosmosSourceConfig.getConfigDef();
+        defineTaskUnitsConfig(configDef);
+
+        return configDef;
+    }
+
+    private static void defineTaskUnitsConfig(ConfigDef result) {
+        //TODO: seems like all the configs does not provide default value will need to be configured, re-evaluate
+        result
+            .defineInternal(
+                SOURCE_FEED_RANGE_TASK_UNITS,
+                ConfigDef.Type.STRING,
+                ConfigDef.NO_DEFAULT_VALUE,
+                ConfigDef.Importance.HIGH
+            )
+            .defineInternal(
+                SOURCE_METADATA_TASK_UNIT,
+                ConfigDef.Type.STRING,
+                null,
+                ConfigDef.Importance.HIGH
+            );
+    }
+
+    private List<FeedRangeTaskUnit> parseFeedRangeTaskUnits() {
+        try {
+            String feedRangesTaskUnitsConfig = this.getString(SOURCE_FEED_RANGE_TASK_UNITS);
+            if (!StringUtils.isEmpty(feedRangesTaskUnitsConfig)) {
+                return OBJECT_MAPPER
+                    .readValue(feedRangesTaskUnitsConfig, new TypeReference<List<String>>() {})
+                    .stream()
+                    .map(taskUnitConfigJson -> {
+                        try {
+                            return OBJECT_MAPPER.readValue(taskUnitConfigJson, FeedRangeTaskUnit.class);
+                        } catch (JsonProcessingException e) {
+                            throw new IllegalArgumentException("Failed to parseFeedRangeTaskUnit", e);
+                        }
+                    })
+                    .collect(Collectors.toList());
+            }
+
+            return new ArrayList<>();
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Failed to parseFeedRangeTaskUnits", e);
```

Comment:
Inlude the string attempted to be parsed as json?

Reply:
good idea, added